### PR TITLE
feat: search productdb

### LINKED ISSUE #103: Include products from product database in search results
Now that we have the product db we should include the product-support pages for products in the search results automatically. Right now, the query-index should contain the product pages directly but for the product-support pages we need to get the product db as part of the search and add matching entries.

### PR DESCRIPTION
**Pull request by karlpauls**
Fix #103 

Test URLs:
- Before: https://main--mammotome--hlxsites.hlx.page/en/
- After: https://103-search-productdb--mammotome--hlxsites.hlx.page/en/

This PR depends on #128 (refactoring to move the product lookup into lib-franklin) - hence, only the blocks/header/* changes are relevant. The other changes can be better discussed in #129 (the PR for #128)

### CODE REVIEW

**Comment by karlpauls** [comment]
@solaris007 - hopefully not too confusing, this PR contains the changes from #129 so you only need to consider the changes to header.js in this PR (assuming we merge the changes in #129 - otherwise, I'll have to adapt this PR).

File: blocks/header/header.js [hunk 1]
```diff
@@ -177,12 +178,30 @@ async function fetchSearchData() {
       throw new Error(`Fetching search data failed with: ${resp.status}`);
     }
   }
-  return window.searchData;
+  return window.searchData.data;
+}
+
+async function fetchProductSupportSearchData() {
+  if (!window.productSearchData) {
+    const language = window.location.pathname.substring(1, window.location.pathname.indexOf('/', 1));
```
Comment (solaris007) [inline blocks/header/header.js:186]: _cosmetics_: i'd cache the location for efficiency and readability:
```
 const languagePath = window.location.pathname;
 const language = languagePath.substring(1, languagePath.indexOf('/', 1));   
```

File: blocks/header/header.js [hunk 2]
```diff
@@ -177,12 +178,30 @@ async function fetchSearchData() {
...
+    const products = await getProducts(language);
+    window.productSearchData = products.flatMap((product) => ([{
```
Comment (solaris007) [inline blocks/header/header.js:188]: _cosmetics_: for readability, you could use destructuring for the prouduct and asset objects:

```
     window.productSearchData = products.flatMap(({Name, Description, ProductCodes, assets}) => ([
       {
         title: Name,
         description: Description,
         path: `/${language}/product-support/${ProductCodes.split('|')[0]}`,
       },
       ...assets.map(({Name, Description, URL}) => ({
         title: Name,
         description: Description,
         path: URL,
       }))
     ])); 
```

**Review by solaris007 — APPROVED** [review]
LGTM, just cosmetics

**Review by solaris007 — APPROVED** [review]

**Pull request merged**